### DISCUSSION TITLE
请问一下 cuda 的 add_cuflags 在哪里实现

### DISCUSSION THREAD

**Discussion by davidxiaozhi** [xmake-io/xmake#3515] [Q&A]
@waruqi 
如果自定义一个新的类似 cuda nvcc 的编译器,如何实现类似add_cuflags 的功能,比如 add_xxflags

**Comment by xq114** [comment]
这个得用custom rule，然后 `add_files("xxx", {myflag = "abc"})` 里面给rule传递flag，rule里面用 `target:fileconfig("myflag")` 获取。像 add_xxflags 这种接口只对最常见的语言做适配，不可能给每个自定义编译器都设置单独接口的

**Reply by davidxiaozhi** [reply]
如果可以,是否可以给出官方add_cuflags的参考案例

**Reply by xq114** [reply]
参考案例？ https://xmake.io/#/manual/project_target?id=targetadd_cuflags 是指这个吗

**Comment by davidxiaozhi** [comment]
不是, add_cuflags  逻辑官方实现,或者类似的 xmake 对应的实现demo

**Reply by xq114** [reply]
https://github.com/xmake-io/xmake/blob/master/xmake/rules/cuda/xmake.lua
这是官方实现
https://github.com/xmake-io/xmake-repo/blob/master/packages/s/suitesparse/port/xmake.lua
https://github.com/xmake-io/xmake-repo/blob/master/packages/l/luajit/port/xmake.lua
自定义规则可以参考这两个rule
